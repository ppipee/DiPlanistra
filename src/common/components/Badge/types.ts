export enum BadgeVariant {
	Outlined = 'outlined',
	Filled = 'filled',
}

export type BadgeVariantTypes = BadgeVariant | 'filled' | 'outlined'

export interface BadgeProps {
	$variant?: BadgeVariantTypes
	$color: string
	$secondColor?: string
}
