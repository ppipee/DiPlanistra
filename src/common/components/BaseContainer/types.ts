import { ScreenName } from 'common/styles/utils/viewport/screenSizes'

export interface BaseContainerProps {
	$screen?: ScreenName
	$padding?: string
	$paddingMobile?: string
	$margin?: string
	$marginMobile?: string
}
