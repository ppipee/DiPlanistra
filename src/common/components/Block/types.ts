export interface BlockProps {
	$padding?: string
	$borderRadius?: string
	$margin?: string
}
