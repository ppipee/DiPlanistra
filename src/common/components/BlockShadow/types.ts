export enum Shadow {
	Default = 'default',
	Hard = 'hard',
}

type ShadowType = Shadow | Shadow.Default | Shadow.Hard

export interface BlockShadowProps {
	$variant?: ShadowType
}
