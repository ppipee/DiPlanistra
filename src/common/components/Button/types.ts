import { ButtonHTMLAttributes } from 'react'

export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
	$size?: SizeType
	$variant?: VariantType
	$responsive?: boolean
	$disable?: boolean
	$color: string
	$secondaryColor?: string
	$shadow?: boolean
	$border?: BorderType
}

type SizeType = typeof ButtonSizes[keyof typeof ButtonSizes]

export const ButtonSizes = {
	Small: 'small',
	Default: 'default',
} as const

type VariantType = typeof ButtonVariants[keyof typeof ButtonVariants]

export const ButtonVariants = {
	Outlined: 'outlined',
	Filled: 'filled',
	Text: 'text',
} as const

export const ButtonBorders = {
	Default: 'default',
	Curve: 'curve',
	Circle: 'circle',
} as const

type BorderType = typeof ButtonBorders[keyof typeof ButtonBorders]
