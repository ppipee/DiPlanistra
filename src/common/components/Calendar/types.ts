export interface CalenderDateProps {
	startDate?: Date
	endDate?: Date
}

export interface CalendarDisplayStateProps {
	year: number
	month: number
}

export interface CalendarDisplayRange {
	startRange?: CalendarDisplayStateProps
	endRange?: CalendarDisplayStateProps
}

export interface CalendarCellProps {
	$inBetween?: boolean
	$selected?: boolean
	$variant?: CalendarCellVariant
	$isFirstCell?: boolean
	$isLastCell?: boolean
	$isDateSelectable?: boolean
}

export type CalendarCellVariant = 'currentMonth' | 'otherMonth' | 'currentDate'
