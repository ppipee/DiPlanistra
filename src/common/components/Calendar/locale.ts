export const CALENDAR_START_DATE_TITLE = ['เริ่มต้น', 'Start Date']
export const CALENDAR_END_DATE_TITLE = ['สิ้นสุด', 'End Date']
