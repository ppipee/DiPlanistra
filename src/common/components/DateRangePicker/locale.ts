export const PICKER_YESTERDAY = ['เมื่อวาน', 'Yesterday']
export const PICKER_LAST_7_DAYS = ['7 วันที่แล้ว', 'Last 7 Days']
export const PICKER_LAST_14_DAYS = ['14 วันที่แล้ว', 'Last 14 Days']
export const PICKER_LAST_30_DAYS = ['30 วันที่แล้ว', 'Last 30 Days']
export const PICKER_THIS_MONTH = ['เดือนนี้', 'This Month']
export const CUSTOM_PICKER = ['กำหนดเอง', 'Custom']
