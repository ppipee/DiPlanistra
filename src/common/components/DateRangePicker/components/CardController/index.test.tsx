import React from 'react'

import { render, fireEvent } from '@testing-library/react'

import { createI18nWrapper } from 'core/locale/testUtils'
import { Locale } from 'core/locale/types'

import * as CardControllerType from '.'

describe('Test CardController', () => {
	const MOCK_DATE = {
		startDate: new Date(2020, 5, 1),
		endDate: new Date(2020, 5, 5),
	}
	const MOCK_CONTROLLER_INDEX = 1

	const selectDateRangeSpy = jest.fn()
	const useDateRangeContextSpy = jest.fn()
	const selectControllerSpy = jest.fn()
	const setDateSpy = jest.fn()

	jest.doMock('common/components/DateRangePicker/context', () => ({
		useDateRangeContext: useDateRangeContextSpy,
	}))
	jest.doMock('common/components/DateRangePicker/utils/selectDateRange', () => selectDateRangeSpy)

	const { default: CardController } = require('.') as typeof CardControllerType

	beforeEach(() => {
		selectDateRangeSpy.mockReturnValue({})
		useDateRangeContextSpy.mockReturnValue({
			date: { ...MOCK_DATE },
			setDate: setDateSpy,
			selectController: selectControllerSpy,
		})
	})

	afterEach(() => {
		jest.resetAllMocks()
	})

	it('should render CardController with TH text correctly', () => {
		const { getByText } = render(
			<CardController name={['เมื่อวานนี้', 'Yesterday']} type="month" index={MOCK_CONTROLLER_INDEX} />,
			{
				wrapper: createI18nWrapper(Locale.th),
			},
		)

		expect(getByText('เมื่อวานนี้')).toBeDefined()
	})

	it('should render CardController with EN text correctly', () => {
		const { getByText } = render(
			<CardController name={['เมื่อวานนี้', 'Yesterday']} type="month" index={MOCK_CONTROLLER_INDEX} />,
			{
				wrapper: createI18nWrapper(Locale.en),
			},
		)

		expect(getByText('Yesterday')).toBeDefined()
	})

	it('should perform selectRange correctly', () => {
		selectDateRangeSpy.mockReturnValue({
			startDate: new Date(2020, 5, 2),
			endDate: new Date(2020, 5, 5),
		})

		const { container } = render(
			<CardController name={['เมื่อวานนี้', 'Yesterday']} type="month" index={MOCK_CONTROLLER_INDEX} />,
			{
				wrapper: createI18nWrapper(Locale.th),
			},
		)

		fireEvent.click(container.firstChild)

		expect(setDateSpy).toBeCalledTimes(1)
		expect(setDateSpy).toBeCalledWith({
			startDate: new Date(2020, 5, 2),
			endDate: new Date(2020, 5, 5),
		})

		expect(selectControllerSpy).toBeCalledTimes(1)
	})

	it('should render CardController with active style', () => {
		selectDateRangeSpy.mockReturnValue({ ...MOCK_DATE })

		const { container } = render(
			<CardController name={['เมื่อวานนี้', 'Yesterday']} type="month" index={MOCK_CONTROLLER_INDEX} />,
			{
				wrapper: createI18nWrapper(Locale.th),
			},
		)

		expect(selectControllerSpy).toBeCalledTimes(1)
		expect(container.firstChild).toMatchSnapshot()
	})

	it('should render CardController without active style', () => {
		selectDateRangeSpy.mockReturnValue({})

		const { container } = render(
			<CardController name={['เมื่อวานนี้', 'Yesterday']} type="month" index={MOCK_CONTROLLER_INDEX} />,
			{
				wrapper: createI18nWrapper(Locale.th),
			},
		)

		expect(selectControllerSpy).not.toBeCalled()
		expect(container.firstChild).toMatchSnapshot()
	})
})
