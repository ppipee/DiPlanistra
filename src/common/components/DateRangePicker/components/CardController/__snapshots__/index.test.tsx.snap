// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CardController should render CardController with active style 1`] = `
<div
  class="styled__Card-grAINz jLQvDg heading4 align-center"
>
  เมื่อวานนี้
</div>
`;

exports[`Test CardController should render CardController without active style 1`] = `
<div
  class="styled__Card-grAINz idryPy heading4 align-center"
>
  เมื่อวานนี้
</div>
`;
